search/enrich/enrich.md

```diff
@@ -0,0 +1,34 @@
+## Enrich
+
+The `enrich` module enables adding enumerated values to all entries in a pipeline within the query or using resources. `enrich` can be used to annotate data with constant enumerated values, such as "User=admin", in order to simplify visualization and reporting, pivoting within compound queries, and working with nontemporal data. 
+
+### Supported Options
+
+* `-r`: Optional. The `-r` option requires an argument and specifies the name or UUID of a resource to extract data from.
+* `-o`: Optional. Overwrite any existing enumerated values that are specified. Usually combined with `-r`. 
+
+### Enriching with string constants
+
+The simplest use of `enrich` is to specify enumerated values and data within the query. For example, to add an enumerated value "foo" with the value "bar" to every entry, simply specify the name and content of the enumerated value, seperated by whitespace:
+
+```
+tag=jsondata json val | enrich foo bar | table val foo
+```
+
+You can specify multiple enumerated value pairs as well. For example, to create enumerated values "foo" and "bar", each with different content:
+
+```
+tag=jsondata json val | enrich foo "my data" bar "my other data" | table val foo bar
+```
+
+### Enriching with resources 
+
+The `enrich` module can extract columns from a CSV or lookup table resource. When using resources, only the first row will be used by `enrich`. Columns can be specified, and all columns will be used if none are specified. If a column conflicts with an existing enumerated value, it will only be overwritten if using the `-o` flag. 
+
+For example, to use columns "foo" and "bar" from a resource "data":
+
+```
+tag=jsondata val | enrich -r data foo bar | table val foo bar"
+```
+
+NOTE: Only the first row of resources are used with the `enrich` module.
```

Comment:
Add something like "When using CSV files, note that the first row refers to the first row of actual data, not the row containing column names"?

Reply:
added "(excluding column headers when using CSV resources)"